dds/DCPS/DataReaderImpl.cpp

```diff
@@ -2292,6 +2292,7 @@ void
 DataReaderImpl::instances_liveliness_update(WriterInfo& info,
     const MonotonicTimePoint&)
 {
+  ACE_GUARD(ACE_Recursive_Thread_Mutex, guard, sample_lock_);
   ACE_GUARD(ACE_Recursive_Thread_Mutex, instance_guard, this->instances_lock_);
```

Comment:
```suggestion
  ACE_GUARD(ACE_Recursive_Thread_Mutex, instance_guard, instances_lock_);
```

Reply:
I didn't change that line...

Reply:
I realize that. I'm suggesting you make them match.